src/Search.hs

```diff
@@ -0,0 +1,54 @@
+module Search
+    ( searchBoards
+    ) where
+
+import Control.Monad (mzero)
+import Data.Maybe (fromMaybe)
+import Data.List (filter)
+
+import Data.Aeson
+import qualified Data.ByteString.Lazy as BS
+import Data.Text as T (Text, pack)
```

Comment:
Data.Text is almost always imported `qualified`, because of the function name clashes you've run into:

```haskell
import qualified Data.Text as T
```

Then you don't need to import `Data.List` (it's in the Prelude) and you don't need to limit what you're pulling in from Data.Text.